watchtower/blob/justice_kit.go

```diff
@@ -1,153 +1,269 @@
+	// be used to locate the input on the breach commitment transaction.
+	toRemoteScriptHash, err := input.CommitScriptUnencumbered(
+		toRemotePubKey,
+	)
+	if err != nil {
+		return nil, nil, 0, err
+	}
 
-	return witnessStack, nil
+	toRemoteSig, err := l.commitToRemoteSig.ToSignature()
+	if err != nil {
+		return nil, nil, 0, err
+	}
+
+	witness := make([][]byte, 2)
+	witness[0] = append(toRemoteSig.Serialize(), byte(txscript.SigHashAll))
+	witness[1] = toRemoteScript
+
+	return toRemoteScriptHash, witness, 0, nil
 }
 
 // HasCommitToRemoteOutput returns true if the blob contains a to-remote p2wkh
 // pubkey.
-func (b *JusticeKit) HasCommitToRemoteOutput() bool {
-	return btcec.IsCompressedPubKey(b.CommitToRemotePubKey[:])
+//
+// NOTE: This is part of the JusticeKit interface.
+func (l *legacyJusticeKit) HasCommitToRemoteOutput() bool {
+	return btcec.IsCompressedPubKey(l.commitToRemotePubKey[:])
 }
 
-// CommitToRemoteWitnessScript returns the witness script for the commitment
-// to-remote output given the blob type. The script returned will either be for
-// a p2wpkh to-remote output or an p2wsh anchor to-remote output which includes
-// a CSV delay.
-func (b *JusticeKit) CommitToRemoteWitnessScript() ([]byte, error) {
-	if !btcec.IsCompressedPubKey(b.CommitToRemotePubKey[:]) {
-		return nil, ErrNoCommitToRemoteOutput
-	}
+// SweepAddress returns the sweep address to be used on the justice tx
+// output.
+//
+// NOTE: This is part of the JusticeKit interface.
+func (l *legacyJusticeKit) SweepAddress() []byte {
+	return l.sweepAddress
+}
 
-	// If this is a blob for an anchor channel, we'll return the p2wsh
-	// output containing a CSV delay of 1.
-	if b.BlobType.IsAnchorChannel() {
-		pk, err := btcec.ParsePubKey(b.CommitToRemotePubKey[:])
-		if err != nil {
-			return nil, err
-		}
+// AddToLocalSig adds the to-local signature to the kit.
+//
+// NOTE: This is part of the JusticeKit interface.
+func (l *legacyJusticeKit) AddToLocalSig(sig lnwire.Sig) {
+	l.commitToLocalSig = sig
+}
 
-		return input.CommitScriptToRemoteConfirmed(pk)
-	}
+// AddToRemoteSig adds the to-remote signature to the kit.
+//
+// NOTE: This is part of the JusticeKit interface.
+func (l *legacyJusticeKit) AddToRemoteSig(sig lnwire.Sig) {
+	l.commitToRemoteSig = sig
+}
+
+// PlainTextSize is the size of the encoded-but-unencrypted blob in
+// bytes.
+//
+// NOTE: This is part of the JusticeKit interface.
+func (l *legacyJusticeKit) PlainTextSize() int {
+	return V0PlaintextSize
+}
 
-	return b.CommitToRemotePubKey[:], nil
+// anchorJusticeKit is an implementation of the JusticeKit interface which can
+// be used for backing up commitments of anchor channels. It inherits most of
+// the methods from the legacyJusticeKit and overrides the
+// ToRemoteOutputSpendInfo method since the to-remote output of an anchor
+// output is a P2WSH instead of the P2WPKH used by the legacy channels.
+type anchorJusticeKit struct {
+	*legacyJusticeKit
```

Comment:
We embed a pointer here where the `legacyJusticeKit` embeds the `justiceKitPacketV0` as a value. Is that on purpose and required to "inherit" the methods?

Reply:
good point  - changed it to be a value 👍